### ISSUE TITLE
1.6.6 breaking during fedora packaging

### ISSUE THREAD

**Issue by dankamongmen** [dankamongmen/notcurses#814]
1.6.3 built on Fedora just fine, complete with test suite. 1.6.6, however, is seeing breakage, I believe before we even get into the compilation step. This might be due to recent changes in Fedora's cmake build macro.

**Comment by dankamongmen** [comment]
https://koji.fedoraproject.org/koji/taskinfo?taskID=47577145 here's a recent one.

**Comment by dankamongmen** [comment]
It looks like this might be a dependency issue (from the x86_64 build):

```
Error: 
 Problem: package OpenImageIO-2.1.17.0-1.fc33.x86_64 requires libopencv_core.so.4.3()(64bit), but none of the providers can be installed
  - package OpenImageIO-2.1.17.0-1.fc33.x86_64 requires libopencv_imgproc.so.4.3()(64bit), but none of the providers can be installed
  - package OpenImageIO-2.1.17.0-1.fc33.x86_64 requires libopencv_videoio.so.4.3()(64bit), but none of the providers can be installed
  - package OpenImageIO-devel-2.1.17.0-1.fc33.x86_64 requires libOpenImageIO.so.2.1()(64bit), but none of the providers can be installed
  - package OpenImageIO-devel-2.1.17.0-1.fc33.x86_64 requires libOpenImageIO_Util.so.2.1()(64bit), but none of the providers can be installed
  - package OpenImageIO-devel-2.1.17.0-1.fc33.x86_64 requires OpenImageIO(x86-64) = 2.1.17.0-1.fc33, but none of the providers can be installed
  - package opencv-core-4.3.0-7.fc33.x86_64 requires libgdal.so.27()(64bit), but none of the providers can be installed
  - conflicting requests
  - nothing provides libpoppler.so.94()(64bit) needed by gdal-libs-3.1.2-4.fc33.x86_64
(try to add '--skip-broken' to skip uninstallable packages)
```

**Comment by dankamongmen** [comment]
In a mock for f33, I get this:

```
+ /usr/bin/cmake -S . -B x86_64-redhat-linux-gnu -DCMAKE_C_FLAGS_RELEASE:STRING=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE:STRING=-DNDEBUG -DCMAKE_Fortran_FLAGS_RELEASE:STRING=-DNDEBUG -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON -DCMAKE_INSTALL_PREFIX:PATH=/usr -DINCLUDE_INSTALL_DIR:PATH=/usr/include -DLIB_INSTALL_DIR:PATH=/usr/lib64 -DSYSCONF_INSTALL_DIR:PATH=/etc -DSHARE_INSTALL_PREFIX:PATH=/usr/share -DLIB_SUFFIX=64 -DBUILD_SHARED_LIBS:BOOL=ON -DUSE_MULTIMEDIA=oiio -DDFSG_BUILD=on ..
-- The C compiler identification is GNU 10.1.1
-- The CXX compiler identification is GNU 10.1.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Requested multimedia engine: oiio
-- Requested build mode: RelWithDebInfo
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.7.0") 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Checking for one of the modules 'tinfo>=6.1;ncursesw>=6.1'
-- Checking for module 'OpenImageIO>=2.1'
--   Found OpenImageIO, version 2.1.17.0
-- Looking for qrcodegen/qrcodegen.h
-- Looking for qrcodegen/qrcodegen.h - found
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_CXX_FLAGS_RELEASE
    CMAKE_C_FLAGS_RELEASE
    CMAKE_Fortran_FLAGS_RELEASE
    INCLUDE_INSTALL_DIR
    LIB_INSTALL_DIR
    LIB_SUFFIX
    SHARE_INSTALL_PREFIX
    SYSCONF_INSTALL_DIR


-- Build files have been written to: /builddir/build/BUILD/notcurses-1.6.6/x86_64-redhat-linux-gnu/x86_64-redhat-linux-gnu
+ /usr/bin/make -O -j8 V=1 VERBOSE=1
make: *** No targets specified and no makefile found.  Stop.


RPM build errors:
error: Bad exit status from /var/tmp/rpm-tmp.1YYvEt (%build)
    Bad exit status from /var/tmp/rpm-tmp.1YYvEt (%build)
Finish: rpmbuild notcurses-1.6.6-1.fc33.src.rpm
Finish: build phase for notcurses-1.6.6-1.fc33.src.rpm
ERROR: Exception(rpmbuild/SRPMS/notcurses-1.6.6-1.fc33.src.rpm) Config(fedora-rawhide-x86_64) 1 minutes 31 seconds
INFO: Results and/or logs in: /var/lib/mock/fedora-rawhide-x86_64/result
ERROR: Command failed: 
 # /usr/bin/systemd-nspawn -q -M 0e36ec5732154919aebc610d3515e10d -D /var/lib/mock/fedora-rawhide-x86_64/root -a -u mockbuild --capability=cap_ipc_lock --bind=/tmp/mock-resolv.j0f94_3g:/etc/resolv.conf --bind=/dev/loop-control --bind=/dev/loop0 --bind=/dev/loop1 --bind=/dev/loop2 --bind=/dev/loop3 --bind=/dev/loop4 --bind=/dev/loop5 --bind=/dev/loop6 --bind=/dev/loop7 --bind=/dev/loop8 --bind=/dev/loop9 --bind=/dev/loop10 --bind=/dev/loop11 --console=pipe --setenv=TERM=vt100 --setenv=SHELL=/bin/bash --setenv=HOME=/builddir --setenv=HOSTNAME=mock --setenv=PATH=/usr/bin:/bin:/usr/sbin:/sbin --setenv=PROMPT_COMMAND=printf "\033]0;<mock-chroot>\007" --setenv=PS1=<mock-chroot> \s-\v\$  --setenv=LANG=C.UTF-8 bash --login -c /usr/bin/rpmbuild -bb --target x86_64 --nodeps /builddir/build/SPECS/notcurses.spec

[fedora33vm](1) $ 
```

**Comment by dankamongmen** [comment]
OK, we're now building within an f33 mock once more. We still seem to be having difficulties on koji due to the OpenImageIO problems.

**Comment by dankamongmen** [comment]
OpenImageIO still isn't installing in the f33 mock, due not to libdc1394 problems :/.

**Comment by dankamongmen** [comment]
It looks like there was a new libdc1394 build two days ago to upgrade to 2.2.6. That probably just needs to filter through koji or something.

**Comment by dankamongmen** [comment]
https://bugzilla.redhat.com/show_bug.cgi?id=1860740

**Comment by dankamongmen** [comment]
OpenCV is fixed on Fedora, OpenImageIO is fixed on Fedora, and Notcurses 1.6.10 is fixed on Fedora =]. Built for f33 this morning.

**Comment by dankamongmen** [comment]
https://koji.fedoraproject.org/koji/taskinfo?taskID=48453101
